client/galaxy/scripts/mvc/ui/ui-select-content.js

```diff
@@ -444,11 +448,17 @@ const View = Backbone.View.extend({
         return v.history_content_type == "dataset_collection" ? "hdca" : "hda";
     },
 
-    /** Library datasets are displayed and selected together with history datasets **/
+    /** Library datasets are displayed and selected together with history datasets,
+        Dataset collection elements are displayed together with history dataset collections **/
     _patchValue: function (v) {
-        if (v.src == "ldda") {
-            v.src = "hda";
-            v.origin = "ldda";
+        const patch_to = { ldda: "hda", dce: "hdca" };
```

Comment:
Minor convention note; camelCase in js, not snake_case like python.

Reply:
Thanks, I've updated the commit!